src/main/java/com/amazon/ion/impl/macro/MacroRef.kt

```diff
@@ -10,4 +10,9 @@ sealed interface MacroRef {
     @JvmInline value class ByName(val name: String) : MacroRef
     @JvmInline value class ById(val id: Long) : MacroRef
     // TODO: Since system macros have an independent address space, do we need to have a `SystemById` variant?
+
+    companion object {
+        @JvmStatic
+        fun byId(id: Long): MacroRef = ById(id)
```

Comment:
I couldn't figure out how to instantiate a MacroRef in Java without this. Let me know if I missed something obvious.

Reply:
When I try returning `MacroRef.ById`, then I run into problems with the `EncodingContext` constructor, which accepts `Map<MacroRef, Macro>` and not `Map<Long, Macro>`. I'll leave this as-is for now and we can change it later as time allows, or as performance targets require.